src/main/java/seedu/address/logic/parser/task/SearchTaskCommandParser.java

```diff
@@ -0,0 +1,45 @@
+package seedu.address.logic.parser.task;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.task.TaskCliSyntax.PREFIX_DUE_DATE;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.task.SearchTaskCommand;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.Prefix;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.task.DateTimeDue;
+
+public class SearchTaskCommandParser implements Parser<SearchTaskCommand> {
```

Comment:
might be good to have a java docs header before this!

Reply:
Sure!